src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -141,90 +143,83 @@ public void postSonarQubeReport(PullRequestRef pr, String sonarQubeURL, SonarQub
+  void postCommentPerIssue(PullRequestRef pr, Collection<Issue> issues, StashDiffReport diffReport, StashClient stashClient) throws StashClientException {
 
     // to optimize request to Stash, builds comment match ordered by filepath
     Map<String, StashCommentReport> commentsByFile = new HashMap<>();
     for (Issue issue : issues) {
-      if (commentsByFile.get(issue.getPath()) == null) {
-        StashCommentReport comments = stashClient.getPullRequestComments(pr, issue.getPath());
+      String path = getIssuePath(issue);
+      if (commentsByFile.get(path) == null) {
+        StashCommentReport comments = stashClient.getPullRequestComments(pr, path);
 
         // According to the type of the comment
         // if type == CONTEXT, comment.line is set to source line instead of destination line
         comments.applyDiffReport(diffReport);
 
-        commentsByFile.put(issue.getPath(), comments);
+        commentsByFile.put(path, comments);
       }
     }
 
-    try {
-      // Severity available to create a task
-      List<String> taskSeverities = getReportedSeverities();
-
-      // Let's call this "issue_loop"
-      for (Issue issue : issues) {
-        StashCommentReport comments = commentsByFile.get(issue.getPath());
-
-        // if comment not already pushed to Stash
-        if ((comments != null) &&
-                (comments.contains(issue.printIssueMarkdown(sonarQubeURL), issue.getPath(), issue.getLine()))) {
-          LOGGER.debug("Comment \"{}\" already pushed on file {} ({})", issue.getKey(),
-                  issue.getPath(), issue.getLine());
-          continue;  // Next element in "issue_loop"
-        }
+    // Severity available to create a task
+    List<String> taskSeverities = getReportedSeverities();
 
-        // check if issue belongs to the Stash diff view
-        String type = diffReport.getType(issue.getPath(), issue.getLine());
-        if (type == null) {
-          LOGGER.info("Comment \"{}\" cannot be pushed to Stash like it does not belong to diff view - {} (line: {})",
-                  issue.getKey(), issue.getPath(), issue.getLine());
-          continue;  // Next element in "issue_loop"
-        }
+    for (Issue issue : issues) {
+        postIssueComment(pr, issue, commentsByFile, diffReport, stashClient, taskSeverities);
+    }
+  }
+
+  private void postIssueComment(PullRequestRef pr, Issue issue, Map<String, StashCommentReport> commentsByFile,
+                                StashDiffReport diffReport, StashClient stashClient, List<String> taskSeverities)
+                                throws StashClientException {
+    String path = getIssuePath(issue);
+    StashCommentReport comments = commentsByFile.get(path);
+    String commentContent = MarkdownPrinter.printIssueMarkdown(issue, config.getSonarQubeURL());
+    Integer issueLine = issue.line();
+    // FIXME move this somewhere else
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)